app/controllers/provider/admin/messages/outbox_controller.rb

```diff
@@ -64,10 +68,22 @@ def recipient_ids
   end
 
   def mass_message?
-    current_account.provider? && params[:to].blank?
+    current_account.provider? && to_param.blank?
   end
 
   def message_params
-    params.fetch(:message, {}).merge(:origin => "web")
+    params.require(:message).permit(:subject, :body).merge(:origin => "web")
+  end
+
+  def message_id_param
+    params.require(:id)
+  end
+
+  def pagination_params
+    { page: params.permit(:page)[:page] }
+  end
+
+  def to_param
+    params.permit(:to).fetch(:to)
```

Comment:
Using `#fetch` will raise a `KeyError` if it is not there. How do you rescue that? We do not want to display a 5xx to the user

Reply:
does original approach where we use `params[:to].blank?` inside `@mass_message?` fail for whatever reason?

Reply:
@hallelujah `:to` should always be there, tests will fail otherwise. In any case, it should fail gracefully:

![Screenshot 2021-12-23 at 09 18 13](https://user-images.githubusercontent.com/11672286/147210444-abf8a69c-1e59-41c0-8e30-efb52ebc2007.png)

@akostadinov I don't think so, do you think it's wrong to permit it first?

Reply:
Removed `param_to`.